### ISSUE TITLE
Calculate Chunk Root Using Blob Serialization & Proof of Custody

### ISSUE THREAD

**Issue by rauljordan** [prysmaticlabs/geth-sharding#112]
Hey all,

To continue with the work done on #100 and #92 with respect to creating a `Shard` struct and the blob serialization implementation done by @nisdas, we now need to calculate ChunkRoot according to the serialization algorithm on the body of a collation.

# Context

In #100, I included a simple `CalculateChunkRoot` method on the `Collation` type. Given this was out of scope of my PR and blocked by #92 being merged,  I chose to have the chunk root be a simple, sha3 hash of the collation body. We need to open a PR that uses the blob serialization to modify the following function created in #100:

```go
// CalculateChunkRoot updates the collation header's chunk root based on the body.
func (c *Collation) CalculateChunkRoot() {
	// TODO: this needs to be based on blob serialization.
	// For proof of custody we need to split chunks (body) into chunk + salt and
	// take the merkle root of that.
	chunkRoot := common.BytesToHash(c.body)
	c.header.data.ChunkRoot = &chunkRoot
}
```

Additionally, for [proof of custody](https://ethresear.ch/t/extending-skin-in-the-game-of-notarization-with-proofs-of-custody/1639/4) (See that ETHResearch post for context), we will need to split the chunks of the body into chunk + salts and take the merkle root of those combined values. This proof of custody mechanism would add more skin-in-the-game for the notarization scheme, preventing them from being lazy when voting on collations.
